modular_skyrat/modules/modular_implants/code/nifs.dm

```diff
@@ -0,0 +1,511 @@
+	if(calibrating)
+		preform_calibration()
+
+	if(IS_IN_STASIS(linked_mob))
+		return
+
+	if((linked_mob.nutrition < minimum_nutrition) && (nutrition_drain)) //Turns nutrition drain off if nutrition is lower than minimum
+		toggle_nutrition_drain(TRUE)
+
+	if(blood_drain && (!blood_check())) //Disables blood draining if the mob fails the blood check
+		toggle_blood_drain(TRUE)
+
+	if(blood_drain)
+		linked_mob.blood_volume -= blood_drain_rate
+
+	if(power_usage > power_level)
+		for(var/datum/nifsoft/nifsoft as anything in loaded_nifsofts)
+			if(!nifsoft.active)
+				continue
+
+			nifsoft.activate()
+
+	power_level += -power_usage
+	if(power_level > max_power)
+		power_level = max_power //No Overcharging
+
+/obj/item/organ/internal/cyberimp/brain/nif/Destroy()
+	linked_mob.nif_examine_text = null
+	linked_mob = null
+
+	QDEL_LIST(loaded_nifsofts)
+	return ..()
+
+/// Subtracts from the power level of the NIF once, this is good for anything that is single use.
+/obj/item/organ/internal/cyberimp/brain/nif/proc/use_power(power_to_use)
+	if(power_level < power_to_use)
+		return FALSE
+
+	power_level -= power_to_use
+	return TRUE
+
+///Toggles nutrition drain as a power source on NIFs on/off
+/obj/item/organ/internal/cyberimp/brain/nif/proc/toggle_nutrition_drain(bypass = FALSE)
+	if(!nutrition_check() && !bypass)
+		return FALSE
+
+	var/hunger_modifier = linked_mob.physiology.hunger_mod
+
+	if(nutrition_drain)
+		hunger_modifier = nutrition_drain_rate
+		power_usage += (nutrition_drain_rate * nutrition_conversion_rate)
+		nutrition_drain = FALSE
+		return TRUE
+
+	hunger_modifier *= nutrition_drain_rate
+	power_usage -= (nutrition_drain_rate * nutrition_conversion_rate)
+	nutrition_drain = TRUE
+
+/// Checks to see if the mob has a nutrition that can be drain from
+/obj/item/organ/internal/cyberimp/brain/nif/proc/nutrition_check() //This is a seperate proc so that TGUI can preform this check on the menu
+	if(!linked_mob || !linked_mob.nutrition)
+		return FALSE
+
+	if(isrobotic(linked_mob))
+		return FALSE
+
+	if(HAS_TRAIT(linked_mob, TRAIT_NOHUNGER)) //Hemophages HATE this one simple check.
+		return FALSE
+
+	if((linked_mob.nutrition < minimum_nutrition)) //No reason why you should be able to turn this on without power
+		return FALSE
+
+	return TRUE
+
+///Toggles Blood Drain
+/obj/item/organ/internal/cyberimp/brain/nif/proc/toggle_blood_drain(bypass = FALSE)
+	if(!blood_check() || bypass)
+		return
+
+	if(blood_drain)
+		blood_drain = FALSE
+		power_usage += (blood_drain_rate * blood_conversion_rate)
+
+		to_chat(linked_mob, span_notice("Blood draining is now disabled"))
+		return
+
+	blood_drain = TRUE
+	power_usage -= (blood_drain_rate * blood_conversion_rate)
+
+	to_chat(linked_mob, span_notice("Blood draining is now enabled."))
+
+///Can we take blood from the mob?
+/obj/item/organ/internal/cyberimp/brain/nif/proc/blood_check()
+	if(!linked_mob || !linked_mob.blood_volume)
+		return FALSE
+
+	if(linked_mob.blood_volume < minimum_blood_level)
```

Comment:
This means that it will allow it at the `minimum_blood_level`, which will allow the user to fall below the `minimum_blood_level`.

Reply:
Also put this on one line, on the return, instead.